### PR TITLE
option to not perform git checkout on SIRF and STIR folders

### PR DESCRIPTION
**Pull request by rijobro**

### CODE REVIEW

**Comment by rijobro** [comment]
@paskino Is this what you meant? Seems to work for me.
If folder exists, give the user the option to not perform git checkout. If true set `GIT_TAG` and `GIT_URL` to blank.

**Comment by rijobro** [comment]
@paskino, @evgueni-ovtchinnikov have you managed to give this a try yet? works very well on my computer so I'm tempted to merge. Any objections @KrisThielemans?

**Comment by rijobro** [comment]
ok, used a macro to avoid duplication.

File: SuperBuild/External_SIRF.cmake [hunk 1]
```diff
@@ -65,10 +65,17 @@ if(NOT ( DEFINED "USE_SYSTEM_${externalProjName}" AND "${USE_SYSTEM_${externalPr
     set(extra_args "")
   endif()
 
+  # Sets ${proj}_URL_MODIFIED and ${proj}_TAG_MODIFIED
+  # If the user doesn't want git checkout to be performed, 
+  # these will be set to blank strings. Else, they'll be set to 
+  # ${${proj}_URL} and ${${proj}_TAG}, respectively.
+  include(${CMAKE_SOURCE_DIR}/cmake/SetGitTagAndRepo.cmake)
```
Comment (KrisThielemans) [inline SuperBuild/External_SIRF.cmake]: Move this and the above lines to SuperBuild.cmake. Even less repetition!

File: CMake/SetGitTagAndRepo.cmake [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# Sets ${proj}_URL_MODIFIED and ${proj}_TAG_MODIFIED
+# If the user doesn't want git checkout to be performed, 
+# these will be set to blank strings. Else, they'll be set to 
+# ${${proj}_URL} and ${${proj}_TAG}, respectively.
+#
+# Example usage:
+#   cmake . -DDISABLE_GIT_CHECKOUT_STIR=ON
+#
+# This will call SetGitTagAndRepo(STIR), setting 
+# STIR_TAG_MODIFIED and STIR_TAG_MODIFIED accordingly.
```
Comment (KrisThielemans) [inline CMake/SetGitTagAndRepo.cmake]:
```suggestion
# STIR_URL_MODIFIED and STIR_TAG_MODIFIED accordingly.
```

File: CMake/SetGitTagAndRepo.cmake [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#=========================================================================
+
+# Sets ${proj}_URL_MODIFIED and ${proj}_TAG_MODIFIED
```
Comment (KrisThielemans) [inline CMake/SetGitTagAndRepo.cmake]: I'd rephrase this a bit now.

`This file defines a macro ...`

**Comment by rijobro** [comment]
done
